### Урок 5. Проверка статистических гипотез. P-значения. Доверительные интервалы. A/B-тестирование ###

In [32]:
from scipy import stats
import numpy as np
from math import sqrt

#### 1.Известно, что генеральная совокупность распределена нормально со средним квадратическим отклонением, равным 16. Найти доверительный интервал для оценки математического ожидания a с надежностью 0.95, если выборочная средняя M = 80, а объем выборки n = 256.####

σ = 16; M = 80; n = 256

In [21]:
80 + stats.norm.ppf(0.025)*(16/16)

78.04003601545995

In [20]:
80 - stats.norm.ppf(0.025)*(16/16)

81.95996398454005

Доверительный интервал: **[78.04; 81.96]**

#### 2.В результате 10 независимых измерений некоторой величины X, выполненных с одинаковой точностью, получены опытные данные: 6.9, 6.1, 6.2, 6.8, 7.5, 6.3, 6.4, 6.9, 6.7, 6.1 Предполагая, что результаты измерений подчинены нормальному закону распределения вероятностей, оценить истинное значение величины X при помощи доверительного интервала, покрывающего это значение с доверительной вероятностью 0,95.####

In [22]:
a = np.array([6.9, 6.1, 6.2, 6.8, 7.5, 6.3, 6.4, 6.9, 6.7, 6.1])

In [35]:
np.mean(a) + stats.t.ppf(0.025, 10-1)*(np.std(a)/sqrt(10))

6.284064674352047

In [37]:
np.mean(a) - stats.t.ppf(0.025, 10-1)*(np.std(a)/sqrt(10))

6.8959353256479545

Величина X принадлежит интервалу **[6.28; 6.89]**, при доверительной вероятности **0.95**

#### 3.Утверждается, что шарики для подшипников, изготовленные автоматическим станком, имеют средний диаметр 17 мм. Используя односторонний критерий с α=0,05, проверить эту гипотезу, если в выборке из n=100 шариков средний диаметр оказался равным 17.5 мм, а дисперсия известна и равна 4 кв.мм.####

H0: µ0 = µ

H1: µ0 < µ

Kт **?** Кр

In [40]:
Zт = stats.norm.ppf(1-0.05)
Zт

1.6448536269514722

In [42]:
Zр = (17.5 - 17)/(2/sqrt(100))
Zр

2.5

Гипотеза **H1 верна** на уровне значимости α=0,05.

#### 4.Продавец утверждает, что средний вес пачки печенья составляет 200 г. Из партии извлечена выборка из 10 пачек. Вес каждой пачки составляет: 202, 203, 199, 197, 195, 201, 200, 204, 194, 190. Известно, что их веса распределены нормально. Верно ли утверждение продавца, если учитывать, что доверительная вероятность равна 99%?####

H0: µ0 = µ

H1: µ0 =/= µ

Kт **?** Кр

In [46]:
b = np.array([202, 203, 199, 197, 195, 201, 200, 204, 194, 190])

In [49]:
Тр = (np.mean(b) - 200)/(np.std(b)/sqrt(10))
Тр

-1.1227217828476797

In [52]:
Tт = stats.t.ppf(0.005, 10-1)
Tт

-3.24983554401537

Гипотеза **H0 верна** на уровне значимости α=0,01.